docstring standardization

steenzout opened the pull request:
Hello,

  I've noticed the docstrings weren't following any convention so I made some changes and made them consistent.
if you want to follow a specific convention let me know and I'll go through the code again.
  There was some text on the docstrings that was broken in a way that made it difficult to read.
I've added some newlines to help with that.

  I've also added a tox env so you can build the docs using

```
$ tox -e docs -- html
```

  I added some :param on some docstrings and fixed a few complaints from sphinx so the HTML documentation should also look better.

  hope it will be useful.

cheers!

erikbern commented:
LGTM!

Tarrasch commented:
Can you rebase so we can merge?

Tarrasch commented on tox.ini:
```diff
@@ -29,3 +29,17 @@ commands = pep8 --ignore E501 luigi test examples bin
 commands = autopep8 --ignore E501 -a -i -r luigi test examples bin
+
+[testenv:docs]
+# to call this use `tox -e docs -- html`
```
Awesome that you are explicit here :)

Tarrasch commented on tox.ini:
```diff
@@ -29,3 +29,17 @@ commands = pep8 --ignore E501 luigi test examples bin
+changedir = doc
+
+commands =
+    make {posargs}
```
How is `{posargs}` different from `{posargs:}`?

steenzout replied:
@Tarrasch it seems `{posargs: ...}` is used when you want to set defaults to positional arguments.

http://tox.readthedocs.org/en/1.0/example/general.html

erikbern commented:
Will this trigger build failures if the documentation is broken? That would be pretty cool!

steenzout commented:
@Tarrasch I've rebased my changes.

erikbern commented:
I think something went wrong when you rebased – mind trying again?

erikbern commented:
Also "fixed URL" might not be the best commit msg :)

erikbern commented:
other than that, LGTM

steenzout commented:
I'v tweaked the command to generate the docs since it was causing tox to throw a warning.
I've fixed a typo.
I've changed the first line of the commit message.

steenzout commented:
@erikbern it may be possible to have the build break if the docs generation fails.
I've tried to introduce some errors and make it fail but sphinx is only giving me warnings.
if I find a way I'll send another PR.
will that work?

steenzout commented:
@erikbern I thought you wanted me to rebase to squash the number of commits.
what do you mean by rebase, exactly?

steenzout commented:
I think I got it.
pull changes from luigi/master and merge the conflicts?

erikbern commented:
Yeah "git pull --rebase" or "git rebase master"

On Tuesday, February 3, 2015, Pedro <notifications@github.com> wrote:

> I think I got it.
> pull changes from luigi/master and merge the conflicts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/luigi/pull/693#issuecomment-72694822>.
>

steenzout commented:
got it.

steenzout commented:
travis failed build due to 'pep8' not being present.
let me have a look...

steenzout commented:
I figured out that you guys run `autopep8` to fix PEP8 warnings.
I've tweaked it a bit so it doesn't add a new line after class declaration.
I think it doesn't makes sense to have the blank line if the next line is the start of the docstring so I kept in on the `pep8` check (which understands what is going on) but removed it from the `autopep8`.

I figure out how to make build break due to sphinx errors.
I'm preparing another PR for that.

I think the rebase is good.
let me know.

erikbern commented:
LGTM let's merge

Tarrasch commented:
Thank you very much! :)
